src/compiler/commandLineParser.ts

```diff
@@ -914,6 +914,7 @@ namespace ts {
      */
     export function readJsonConfigFile(fileName: string, readFile: (path: string) => string | undefined): JsonSourceFile {
         const textOrDiagnostic = tryReadFile(fileName, readFile);
+        // tslint:disable-next-line no-object-literal-type-assertion (TODO:GH#18217)

```

Comment:
we should change the return type for this function instead.

Reply:
To `{ parseDiagnostics: Diagnostic[] }`? That exposes many errors in its uses because we seem to assume we get the full source file.

Reply:
with optional members for `fileName` and `extendedSourceFiles`.. not sure if we need other things as well.

Reply:
that might end up touching many places though..